lib/bn.js

```diff
@@ -3169,7 +3176,10 @@
 
   Red.prototype.imod = function imod (a) {
     if (this.prime) return this.prime.ireduce(a)._forceRed(this);
-    return a.umod(this.m)._forceRed(this);
+
+    var mod = a.umod(this.m)._forceRed(this);
+    if (mod !== a) mod.move(a);
```

Comment:
is it even worth checking if they are different? (performance diff?)

Reply:
In one case `mod` === `a`. Not sure does it cost check it or not... https://github.com/indutny/bn.js/blob/98ac84e81dc02d178168460e5889c80840853a44/lib/bn.js#L2413

Reply:
right,  if `mod === a`, no need to move,  but,  as `move` guarantees that `a == mod`,  why bother checking at all?
It looks like an optimization,  but,  IMHO serves to confuse.

Reply:
Changed to `a.umod(this.m)._forceRed(this).move(a);`